File: src/musikdroid/app/src/main/java/io/casey/musikcube/remote/InvalidPasswordDialogFragment.java
package io.casey.musikcube.remote;

import android.app.Dialog;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

public class InvalidPasswordDialogFragment extends DialogFragment {
    public static final String TAG = "InvalidPasswordDialogFragment";

    public static InvalidPasswordDialogFragment newInstance() {
        return new InvalidPasswordDialogFragment();
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return new AlertDialog.Builder(getActivity())
            .setTitle(R.string.invalid_password_dialog_title)
            .setMessage(R.string.invalid_password_dialog_message)
            .setPositiveButton(R.string.button_settings, (dialog, which) -> {
                startActivity(SettingsActivity.getStartIntent(getActivity()));
            })
            .setNegativeButton(R.string.button_close, null)
            .create();
    }
}

File: src/musikdroid/app/src/main/java/io/casey/musikcube/remote/ui/home/fragment/InvalidPasswordDialogFragment.kt
package io.casey.musikcube.remote.ui.home.fragment

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment

import io.casey.musikcube.remote.R
import io.casey.musikcube.remote.ui.settings.activity.SettingsActivity

class InvalidPasswordDialogFragment : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val activity = this.activity!!
        return AlertDialog.Builder(activity)
            .setTitle(R.string.invalid_password_dialog_title)
            .setMessage(R.string.invalid_password_dialog_message)
            .setPositiveButton(R.string.button_settings) { _, _ ->
                startActivity(SettingsActivity.getStartIntent(activity))
            }
            .setNegativeButton(R.string.button_close, null)
            .create()
    }

    companion object {
        const val TAG = "InvalidPasswordDialogFragment"

        fun newInstance(): InvalidPasswordDialogFragment {
            return InvalidPasswordDialogFragment()
        }
    }
}
